protostar/testing/cheatcodes/expect_call_cheatcode.py

```diff
@@ -21,31 +28,46 @@ def __init__(
     def name(self) -> str:
         return "expect_call"
 
-    def build(self) -> Callable[[int, str, list[int]], None]:
+    def build(self) -> Callable[[int, str, list[int]], Callable]:
         return self.expect_call
 
     def expect_call(
-        self, contract_address: RawAddress, fn_name: str, calldata: list[int]
-    ) -> None:
-        address = Address.from_user_input(contract_address)
-
-        def callback():
-            exists = False
-            expected_fn_selector = get_selector_from_name(fn_name)
-            call_entries = self.cheatable_state.contract_calls.get(address)
-            if call_entries:
-                for called_fn_selector, call_entry in call_entries:
-                    if (
-                        call_entry == calldata
-                        and expected_fn_selector == called_fn_selector
-                    ):
-                        exists = True
-                        break
-            if not exists:
+        self, raw_address: RawAddress, fn_name: str, calldata: list[int]
+    ) -> Callable:
+        contract_address = Address.from_user_input(raw_address)
+        selector, calldata = generate_expected_call_calldata(
+            fn_name=fn_name, calldata=calldata
+        )
+
+        if self.cheatable_state.expected_contract_calls.get(contract_address):
+            self.cheatable_state.expected_contract_calls[contract_address].append(
+                (selector, calldata)
+            )
+        else:
+            self.cheatable_state.expected_contract_calls[contract_address] = [
+                (selector, calldata)
+            ]
+
+        def stop_callback():
+            data_for_address = self.cheatable_state.expected_contract_calls.get(
+                contract_address
+            )
+            if data_for_address and (selector, calldata) in data_for_address:
+                raise Exception("expected call but it didn't happen todo this msg")
```

Comment:
"... todo this msg"

Reply:
[fix raised error](https://github.com/software-mansion/protostar/pull/1213/commits/3786f2d4059f605e9fd5ea5fa98e611263578a97)